### PR TITLE
Add Flax FAQ - how to search, how to take the derivative w.r.t. a hidden layer, remat_scan vs scan(remat), recommended training libraries/metrics

### PR DESCRIPTION
**Pull request by 8bitmp3**
This PR adds a new Flax FAQ.
- Item 1: How to search
- Item 2: How to take a derivative w.r.t. a hidden layer with `flax.linen.Module.perturb`
- Item 3: Flax Linen `remat_scan` vs. `scan(remat(...))`
- Item 4: Recommended training loop libraries (CLU, etc)

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Where to search for an answer to a Flax-related question
+********************************************************
+
+There are a number of official Flax resources to search for information:
+
+- `Flax Documentation on ReadTheDocs <https://github.com/google/flax/discussions>`__ (this site): Use the search bar or the table of contents on the left-hand side.
+- `google/flax GitHub Discussions <https://github.com/google/flax/discussions>`__: Search for an existing topic or start a new one. If you can't find what you're looking for, feel free to ask the Flax team or community a question.
+- `google/flax GitHub Issues <https://github.com/google/flax/discussions>`__.
+
+How to take the derivative with respect to a hidden layer values (using :code:`Module.perturb`)
```
Comment (cgarciae) [inline docs/faq.rst:15]:
```suggestion
How to take the derivative with respect to an intermediate value (using :code:`Module.perturb`)
```

File: docs/faq.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+For full examples and detailed documentation, go to:
+
+- The :meth:`flax.linen.Module.perturb` API docs
+- The `Extracting gradients of intermediate values <https://flax.readthedocs.io/en/latest/guides/extracting_intermediates.html#extracting-gradients-of-intermediate-values>`_ guide
+- `Flax GitHub Discussions #1152 <https://github.com/google/flax/discussions/1152>`_
+
+Is Flax Linen :code:`remat_scan()` the same as :code:`scan(remat(...))`?
+************************************************************************
+
+Flax :code:`remat_scan()` (:meth:`flax.linen.remat_scan()`) and :code:`scan(remat(...))` (:meth:`flax.linen.scan` over :meth:`flax.linen.remat`) are not the same, and :code:`remat_scan()` is limited in cases it supports.
```
Comment (cgarciae) [inline docs/faq.rst:29]: Here we should clarify what "is limited in cases it supports" means, specifically that `remat_scan` treats the inputs and outputs as carries (there are no scan dimension for the data). This makes it easy to implement the "scan over layers" pattern for an existing Module as shown in the docs for `remat_scan`, with `scan(remat(...))` you have create a wrapper Module to handle the scan inputs/outputs.

File: docs/faq.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+================================
+
+This is a collection of answers to frequently asked questions (FAQ). You can contribute to the Flax FAQ by starting a new topic in `GitHub Discussions <https://github.com/google/flax/discussions>`__.
+
+Where to search for an answer to a Flax-related question
+********************************************************
+
+There are a number of official Flax resources to search for information:
+
+- `Flax Documentation on ReadTheDocs <https://github.com/google/flax/discussions>`__ (this site): Use the search bar or the table of contents on the left-hand side.
```
Comment (chiamp) [inline docs/faq.rst:11]:
```suggestion
- `Flax Documentation on ReadTheDocs <https://flax.readthedocs.io/en/latest/>`__ (this site): Use the `search bar <https://flax--3267.org.readthedocs.build/en/3267/search.html>` or the table of contents on the left-hand side.
```

File: docs/faq.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- `google/flax GitHub Discussions <https://github.com/google/flax/discussions>`__: Search for an existing topic or start a new one. If you can't find what you're looking for, feel free to ask the Flax team or community a question.
+- `google/flax GitHub Issues <https://github.com/google/flax/discussions>`__.
```
[the remark this answers is not in the document]
Comment (chiamp) [reply docs/faq.rst:13]: Also maybe add a brief description for searching in Github Issues

File: docs/faq.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- `Flax GitHub Discussions #1152 <https://github.com/google/flax/discussions/1152>`_
+
+Is Flax Linen :code:`remat_scan()` the same as :code:`scan(remat(...))`?
+************************************************************************
+
+Flax :code:`remat_scan()` (:meth:`flax.linen.remat_scan()`) and :code:`scan(remat(...))` (:meth:`flax.linen.scan` over :meth:`flax.linen.remat`) are not the same, and :code:`remat_scan()` is limited in cases it supports.
+
+You are recommended to use :code:`scan(remat(...))`, as typically you would need the extra parameters, such as ``in_axes`` or ``out_axes``, which :meth:`flax.linen.remat_scan` does not expose.
+
+References:
```
Comment (chiamp) [inline docs/faq.rst:33]: Might not need these links below since they were already mentioned previously

File: docs/faq.rst (hunk 3 above)
Comment (8bitmp3) [reply docs/faq.rst:11]: :+1: 
Nit: https://flax.readthedocs.io/en/latest/search.html

File: docs/faq.rst [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+************************************************************************
+
+Flax :code:`remat_scan()` (:meth:`flax.linen.remat_scan()`) and :code:`scan(remat(...))` (:meth:`flax.linen.scan` over :meth:`flax.linen.remat`) are not the same, and :code:`remat_scan()` is limited in cases it supports. Namely, :code:`remat_scan()` treats the inputs and outputs as carries (hidden states that are carried through the training loop). You are recommended to use :code:`scan(remat(...))`, as typically you would need the extra parameters, such as ``in_axes`` (for input array axes) or ``out_axes`` (output array axes), which :meth:`flax.linen.remat_scan` does not expose.
+
+What are the recommended training loop libraries?
+*************************************************
+
+Consider using CLU (Common Loop Utils) `google/CommonLoopUtils <https://github.com/google/CommonLoopUtils>`__. To get started, go to this `CLU Synopsis Colab <https://colab.research.google.com/github/google/CommonLoopUtils/blob/main/clu_synopsis.ipynb>`__. You can find answers to common questions about CLU with Flax on `google/flax GitHub Discussions <https://github.com/google/flax/discussions>`__.
```
Comment (chiamp) [inline docs/faq.rst:34]:
```suggestion
Consider using CLU (Common Loop Utils) `google/CommonLoopUtils <https://github.com/google/CommonLoopUtils>`__. To get started, go to this `CLU Synopsis Colab <https://colab.research.google.com/github/google/CommonLoopUtils/blob/main/clu_synopsis.ipynb>`__. You can find answers to common questions about CLU with Flax on `google/flax GitHub Discussions <https://github.com/google/flax/discussions?discussions_q=clu>`__.
```
Maybe link it to discussions filtered for CLU?

**Review by chiamp — APPROVED** [review]

**Pull request merged**